Framework/Algorithms/test/FilterEventsTest.h

```diff
@@ -91,14 +93,79 @@ class FilterEventsTest : public CxxTest::TestSuite {
     return;
   }
 
-  //----------------------------------------------------------------------------------------------
+  void test_protonCharge() {
+    std::cout << std::endl << "test_protonCharge..." << std::endl;
+    int64_t runstart_i64 = 10000000000; // 10 seconds, beginning of the fake run
+    int64_t pulsedt = 10000000;         // 1/100 seconds in nanoseconds, time between consecutive pulses
+    int64_t tofdt = 1000000;            // 1 milisecond in nanoseconds, spacing between neutrons events within a pulse
+    size_t numpulses = 1000;
+
+    // Create EventWorkspace with 10 detector-banks, each bank containing one pixel.
+    // the total proton charge is 1000
+    EventWorkspace_sptr inpWS = createEventWorkspace(runstart_i64, pulsedt, tofdt, numpulses);
+    AnalysisDataService::Instance().addOrReplace("TestProtonCharge", inpWS);
+
+    // Create SplittersWorkspace with 999 intervals and three output workspaces cycling every three pulses
+    // Notice below that the proton charge of the first pulse goes to the unfiltered workspace
+    // index   DateAndTime
+    //   0     1990-Jan-01 00:00:10.005000000
+    //   1     1990-Jan-01 00:00:10.015000000
+    //   2     1990-Jan-01 00:00:10.025000000
+    //   0     1990-Jan-01 00:00:10.035000000
+    //   1     1990-Jan-01 00:00:10.045000000
+    //   2     1990-Jan-01 00:00:10.055000000
+    //   0     1990-Jan-01 00:00:10.065000000
+    //...
+    //   2     1990-Jan-01 00:00:19.985000000
+    //  -1     1990-Jan-01 00:00:19.995000000
+    std::string name{"splitterPeriodic"};
+    int64_t start{runstart_i64 + 5000 * 1000}; // an offset of 5 miliseconds, in nanoseconds
+    int64_t end{runstart_i64 + pulsedt * static_cast<int64_t>(numpulses)};
+    int64_t splitInterval{pulsedt};
+    int splitterCount{3}; // number of destination workspaces
+    SplittersWorkspace_sptr splittersWorkspace = createPeriodSplitters(start, end, splitInterval, splitterCount);
+    AnalysisDataService::Instance().addOrReplace("SplitterPeriodic", splittersWorkspace);
+
+    FilterEvents filter;
+    filter.initialize();
+
+    // Set properties
+    filter.setProperty("InputWorkspace", "TestProtonCharge");
+    filter.setProperty("OutputWorkspaceBaseName", "PChargeSplit");
+    filter.setProperty("SplitterWorkspace", "SplitterPeriodic");
+    filter.setProperty("OutputTOFCorrectionWorkspace", "PChargeCorrection");
+    filter.setProperty("OutputUnfilteredEvents", true);
+
+    // Execute
+    TS_ASSERT_THROWS_NOTHING(filter.execute());
+    TS_ASSERT(filter.isExecuted());
+
+    // Get output
+    int numsplittedws = filter.getProperty("NumberOutputWS");
+    TS_ASSERT_EQUALS(numsplittedws, 4);
+
+    std::vector<std::string> suffixes{"0", "1", "2", "unfiltered"};
+    std::vector<double> protonCharges;
+    for (const std::string suffix : suffixes) {
```

Comment:
The osx warning created can be solved by making this
```suggestion
    for (const std::string &suffix : suffixes) {
```